File: artifacts/api-server/src/bot/util/embeds.ts
import { EmbedBuilder, type ColorResolvable } from "discord.js";

const PRIMARY = 0xe50914;
const SUCCESS = 0x2ecc71;
const ERROR = 0xe74c3c;
const INFO = 0x3498db;

export function primaryEmbed(title: string, description?: string): EmbedBuilder {
  const e = new EmbedBuilder().setColor(PRIMARY).setTitle(title);
  if (description) e.setDescription(description);
  return e;
}

export function successEmbed(title: string, description?: string): EmbedBuilder {
  const e = new EmbedBuilder().setColor(SUCCESS).setTitle(`✅ ${title}`);
  if (description) e.setDescription(description);
  return e;
}

export function errorEmbed(title: string, description?: string): EmbedBuilder {
  const e = new EmbedBuilder().setColor(ERROR).setTitle(`❌ ${title}`);
  if (description) e.setDescription(description);
  return e;
}

export function infoEmbed(title: string, description?: string): EmbedBuilder {
  const e = new EmbedBuilder().setColor(INFO).setTitle(title);
  if (description) e.setDescription(description);
  return e;
}

export function teamColor(hex: string | null | undefined): ColorResolvable {
  if (!hex) return PRIMARY;
  const cleaned = hex.replace("#", "");
  const parsed = parseInt(cleaned, 16);
  if (Number.isNaN(parsed)) return PRIMARY;
  return parsed as ColorResolvable;
}

File: artifacts/api-server/utils/embeds.py
"""Discord embed yardımcıları."""
import discord


PRIMARY = 0xe50914
SUCCESS = 0x2ecc71
ERROR   = 0xe74c3c
INFO    = 0x3498db


def primary_embed(title: str, description: str | None = None) -> discord.Embed:
    e = discord.Embed(title=title, color=PRIMARY)
    if description:
        e.description = description
    return e


def success_embed(title: str, description: str | None = None) -> discord.Embed:
    e = discord.Embed(title=f"✅ {title}", color=SUCCESS)
    if description:
        e.description = description
    return e


def error_embed(title: str, description: str | None = None) -> discord.Embed:
    e = discord.Embed(title=f"❌ {title}", color=ERROR)
    if description:
        e.description = description
    return e


def info_embed(title: str, description: str | None = None) -> discord.Embed:
    e = discord.Embed(title=title, color=INFO)
    if description:
        e.description = description
    return e


def team_color(hex_color: str | None) -> int:
    if not hex_color:
        return PRIMARY
    cleaned = hex_color.lstrip("#")
    try:
        return int(cleaned, 16)
    except ValueError:
        return PRIMARY
